ios/Sources/MaaS/UI/Components/Badge.swift

```diff
@@ -1,65 +1,309 @@
+        switch input.type {
+        case .small:
+            return constants.textStyleSmall
+        case .medium:
+            return constants.textStyleMedium
+        }
+    }
+
+    private var padding: EdgeInsets {
+        switch input.type {
+        case .small:
+            return .init(top:  constants.verticalPaddingSmall,
+                         leading: constants.horizontalPaddingSmall,
+                         bottom: constants.verticalPaddingSmall,
+                         trailing: constants.horizontalPaddingSmall)
+        case .medium:
+            return .init(top: constants.verticalPaddingMedium,
+                         leading: constants.horizontalPaddingMedium,
+                         bottom: constants.verticalPaddingMedium,
+                         trailing: constants.horizontalPaddingMedium)
+        }
+    }
+
+    private var showText: Bool { input.text?.count ?? 0 > 0 }
+
+}
+
+private struct StackedBadge: View, Swappable {
+
+    struct InputType {
+        let base: BaseBadge.InputType
+        let colors: [Color]
+    }
+
+    let input: InputType
+
+    init(
+        backgroundColor: Color?,
+        icon: UIImage?,
+        text: String?,
+        foregroundColor: Color? = nil,
+        subBadge: UIImage? = nil,
+        isEnabled: Bool = true,
+        stackColors: [Color]
+    ) {
+        input = InputType(
+            base: Badge.InputType(
+                type: .medium,
+                backgroundColor: backgroundColor,
+                foregroundColor: foregroundColor,
+                icon: icon,
+                text: text,
+                alternativeBadgesColors: [],
+                subBadgeIcon: subBadge,
+                isEnabled: isEnabled
+            ),
+            colors: stackColors
+        )
+    }
+
+    @Environment(\.currentTheme) var theme
+
+    var constants: Kotlin<BadgeConstants> { Kotlin(BadgeConstants(theme: theme)) }
+
+    var defaultBody: some View {
+
+        ZStack (alignment: .top) {
+
+            ForEach(0..<badges.count) {
+                BaseBadge(input: badges[$0])
+                    .frame(maxWidth: .infinity)
+                    .withStroke(cornerRadius: constants.cornerRadiusMedium,
+                                fillColor: badges[$0].backgroundColor ?? .clear,
+                                lineWidth: constants.borderWidth)
+                    .padding(.top, spacing(index: $0))
+            }
+        }
+        .fixedSize()
+        .subBadge(image: input.base.subBadgeIcon,
+                  width: constants.subBadgeIconWidth,
+                  height: constants.subBadgeIconHeight)
+    }
+
+    private var badges: [Badge.InputType] {
+        let maxAlternativeBadges = 2
+        let alternativeBadges = Array(input.colors.prefix(maxAlternativeBadges)).toAlternativeBadges()
+        return alternativeBadges + [input.base]
+    }
+
+    private func spacing(index: Int) -> CGFloat {
+        CGFloat(2.5 * Double(index))
```

Comment:
This should probably also be a constant

Reply:
Yes! This actually can be computed from the constants we already have!